tmva/sofie/inc/TMVA/ROperator_Constant.hxx

```diff
@@ -0,0 +1,107 @@
+#ifndef TMVA_SOFIE_ROPERATOR_Constant
+#define TMVA_SOFIE_ROPERATOR_Constant
+
+#include "TMVA/SOFIE_common.hxx"
+#include "TMVA/ROperator.hxx"
+#include "TMVA/RModel.hxx"
+
+#include <sstream>
+
+namespace TMVA{
+namespace Experimental{
+namespace SOFIE{
+
+template<typename T>
+class ROperator_Constant final : public ROperator
+{
+
+private:
+
+   std::string fNX;
+   std::string fNY;
+   std::vector<size_t> fShape;
+   std::vector<T> fValues;
+   std::string fAttrType;
+   bool fIsConstantOfShape = false;
+
+public:
+   ROperator_Constant(){}
+
+   ROperator_Constant(const std::string & type, const std::vector<T> & values, const std::vector<size_t> & shape, std::string nameX, std::string nameY):
+      fNX(UTILITY::Clean_name(nameX)),
+      fNY(UTILITY::Clean_name(nameY)),
+      fShape(shape),
+      fValues(values),
+      fAttrType(type)
+      { }
+
+   std::vector<ETensorType> TypeInference(std::vector<ETensorType> input){
+      return input;
+   }
+
+   std::vector<std::vector<size_t>> ShapeInference(std::vector<std::vector<size_t>> input){
```

Comment:
Is the goal of this function to make a copy of the input? The implementation also is strange, right? Why not just making a copy instead of calling it?